paymentsheet/src/main/java/com/stripe/android/elements/PaymentElementViewModel.kt

```diff
@@ -0,0 +1,165 @@
+package com.stripe.android.elements
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.PaymentIntent
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.forms.FormFieldValues
+import com.stripe.android.paymentsheet.model.PaymentSelection
+import com.stripe.android.paymentsheet.model.getPMAddForm
+import com.stripe.android.paymentsheet.paymentdatacollection.FormFragmentArguments
+import com.stripe.android.ui.core.Amount
+import com.stripe.android.ui.core.FieldValuesToParamsMapConverter
+import com.stripe.android.ui.core.elements.IdentifierSpec
+import com.stripe.android.ui.core.forms.resources.LpmRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.stateIn
+
+/**
+ * ViewModel for the PaymentElement.
+ * Creates the parameters that define the form elements displayed by the [FormViewModel], based on
+ * the payment method type selected. Also transforms the form values into a [PaymentSelection],
+ * which can be used to confirm a payment.
+ */
+internal class PaymentElementViewModel internal constructor(
+    val supportedPaymentMethods: List<LpmRepository.SupportedPaymentMethod>,
+    private val paymentElementConfig: PaymentElementConfig,
+    context: Context,
+    lifecycleScope: CoroutineScope
+) : AndroidViewModel(context.applicationContext as Application) {
+    val selectedPaymentMethod = MutableStateFlow(getInitiallySelectedPaymentMethod())
+
+    private var lastPaymentSelection = paymentElementConfig.initialSelection
+
+    val formArgumentsFlow = selectedPaymentMethod.map { selectedItem ->
+        getFormArgumentsForPaymentMethod(selectedItem)
+    }.stateIn(
+        lifecycleScope,
+        SharingStarted.Eagerly,
```

Comment:
Why do we need to use the `lifecycleScope` here instead of the `viewModelScope`, and why do we need the eager starting?